lib/elixir/lib/access.ex

```diff
@@ -31,83 +26,52 @@ defmodule Access do
       iex> star_ratings[1.5]
       "★☆"
 
-  Note that the dynamic lookup syntax (`term[key]`) roughly translates to
-  `Access.get(term, key, nil)`.
+  The bracket syntax itself places no restrictions on the key types, but the data
+  structure being accessed may have such restrictions (ie. maps can have any key type,
+  but keyword lists must have atoms as keys).
 
-  `Access` can be combined with `Kernel.put_in/3` to put a value
-  in a given key:
+  This works well for dynamic data structures where keys can come and go over the
+  lifetime of a program _but is disabled by default for structs_, which have keys
+  defined at _compile-time_ (see: [Structs and Dot Syntax](#module-structs-and-dot-syntax)).
 
-      iex> map = %{a: 1, b: 2}
-      iex> put_in map[:a], 3
-      %{a: 3, b: 2}
+  ### Bracket Syntax and Access
 
-  This syntax is very convenient as it can be nested arbitrarily:
+  When compiling  source code, Elixir transforms each invocation of bracket syntax
```

Comment:
Yes, good thought. I’d sort of wanted to _prove_ to the reader how that was the case, but maybe some more generic text is sufficient.